src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -491,6 +491,9 @@ public IThreadPoolWorkItem Dequeue(ThreadPoolWorkQueueThreadLocals tl, ref bool
 
         internal static bool Dispatch()
         {
+            // Put threadpool thread on default context
+            ExecutionContextSwitcher.SetDefaults();
```

Comment:
I kinda agree; but the EC mix with default and flow suppressed is more complicated so it would need a greater set of changes.

e.g. For `Capture` for `CurrentThread.ExecutionContext`

`EC == null` => `Default`
`EC != null && FlowSupressed` => `null`
`EC != null` =>`EC`

So `null` means flow suppressed, unless you recapture when it means Default